{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "993af091-22c6-44b6-941d-6ed22f29dc0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import os, sys\n",
    "import seaborn as sns\n",
    "import random\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d1c8388a-9ff9-48d7-8e3e-744406f061ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2995fd24-ee02-42a9-be93-c760eeaa6892",
   "metadata": {},
   "source": [
    "##### Outgroups\n",
    "- outgroup 1:GCF_000423645-1-G568_RS0108480, GCF_000169415-1-SSE37_RS08100, GCF_900110775-1-BSF79_RS18365\n",
    "- outgroup 2: GCF_000421265.1-K355_RS0117460\n",
    "-outgroup 3: GCF_000423645-1-G568_RS0108480, GCF_000169415-1-SSE37_RS08100, GCF_900110775-1-BSF79_RS18365, GCF_000421265.1-K355_RS0117460"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0eeb93c-2a4c-4461-8aad-10c1892a0e69",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Models to deal with dublications of AAA+ domain RRs\n",
    "- outgroup 1\n",
    "    - Model 6: includes ADU59_RS12685 and CEP68_RS17635\n",
    "    - Model 8:includes G568_RS0115095, BSF79_RS15930, SSE37_RS16260, RUTHE_RS05685, BG454_RS09765, K355_RS0117460\n",
    "    - Model 12: includes all AAA+ RRs\n",
    "    - Model 13: includes BG454_RS16400, ADU59_RS19520, K355_RS0119855, B152_RS0108145\n",
    "    - Model 7: excludes GerE\n",
    "    - Model 14: includes BG454_RS16400, ADU59_RS19520, K355_RS0119855, B152_RS0108145 and also includes G568_RS0115095, BSF79_RS15930, SSE37_RS16260, RUTHE_RS05685, BG454_RS09765, K355_RS0117460\n",
    "- outgroup 2\n",
    "    - Model 1: includes ADU59_RS12685 and CEP68_RS17635\n",
    "    - Model 2:includes G568_RS0115095, BSF79_RS15930, SSE37_RS16260, RUTHE_RS05685, BG454_RS09765, K355_RS0117460\n",
    "    - Model 3: includes all AAA+ RRs\n",
    "    - Model 5: includes BG454_RS16400, ADU59_RS19520, K355_RS0119855, B152_RS0108145\n",
    "- outgroup 3\n",
    "    - Model 4: includes all AAA+ RRs\n",
    "    - Model 9: excludes GerE\n",
    "- no outgroup\n",
    "    - Model 10: exlcudes outgroup (use for HyPHY MEME site selection)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "56cf1d55-347e-4c02-ae24-806d6e6fdb8a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GCF_000021325.1-GDIA_RS09815',\n",
       " 'GCF_000423645.1-G568_RS0108480',\n",
       " 'GCF_000169415.1-SSE37_RS08100',\n",
       " 'GCF_900110775.1-BSF79_RS18365',\n",
       " 'GCF_000017565.1-PLAV_RS12075',\n",
       " 'GCF_000017565.1-PLAV_RS12070',\n",
       " 'GCF_000017565.1-PLAV_RS11965',\n",
       " 'GCF_000017565.1-PLAV_RS11955',\n",
       " 'GCF_000017565.1-PLAV_RS10335',\n",
       " 'GCF_000017565.1-PLAV_RS05525',\n",
       " 'GCF_001687365.1-ADU59_RS05750',\n",
       " 'GCF_000385335.1-A3OQ_RS0103935',\n",
       " 'GCF_900168195.1-B5Y19_RS10715',\n",
       " 'GCF_900168195.1-B5Y19_RS10675',\n",
       " 'GCF_900168195.1-B5Y19_RS10665',\n",
       " 'GCF_000374005.1-B152_RS0109090',\n",
       " 'GCF_002208825.2-CEP68_RS27755',\n",
       " 'GCF_000423645.1-G568_RS0115095',\n",
       " 'GCF_900110775.1-BSF79_RS15930',\n",
       " 'GCF_000169415.1-SSE37_RS16260',\n",
       " 'GCF_000442315.1-RUTHE_RS05685',\n",
       " 'GCF_001870665.2-BG454_RS09765',\n",
       " 'GCF_002208825.2-CEP68_RS17635',\n",
       " 'GCF_001687365.1-ADU59_RS12685',\n",
       " 'GCF_001870665.2-BG454_RS16400',\n",
       " 'GCF_001687365.1-ADU59_RS19520',\n",
       " 'GCF_000421265.1-K355_RS0119855',\n",
       " 'GCF_000374005.1-B152_RS0108145']"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model12 = []\n",
    "for seq_record in SeqIO.parse(\"Alphaproteobacteria_subcluster_CDS_REC_model12.fasta\",  \"fasta\"):\n",
    "    model12.append(str(seq_record.id).replace(\"-1\", \".1\").replace(\"-2\", \".2\"))\n",
    "    \n",
    "model12\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "cd75bff2-908a-4934-9ad8-df04d8c47fad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28\n",
      "/opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_AA_model12.fasta \n",
      " /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_AA_model12.afa\n",
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "rescale = 1\n",
      "All-to-all alignment.\n",
      "tbfast-pair (aa) Version 7.419\n",
      "alg=L, model=BLOSUM62, 2.00, -0.10, +0.10, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "rescale = 1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "   20 / 28\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP    13 /27 \n",
      "Reallocating..done. *alloclen = 1235\n",
      "STEP    27 /27 \n",
      "done.\n",
      "tbfast (aa) Version 7.419\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "rescale = 1\n",
      "\n",
      "   20 / 28\n",
      "Segment   1/  1    1- 148\n",
      "STEP 002-001-0  identical.    identical.    identical.    identical.   \n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (aa) Version 7.419\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "model = 12\n",
    "model12 = []\n",
    "records = []\n",
    "for seq_record in SeqIO.parse(f\"Alphaproteobacteria_subcluster_CDS_REC_model{model}.fasta\",  \"fasta\"):\n",
    "    \n",
    "    \n",
    "    seq_record.seq = seq_record.seq.translate()\n",
    "    records.append(seq_record)\n",
    "\n",
    "\n",
    "print(len(records))\n",
    "handle = open(os.path.join(f\"Alphaproteobacteria_subcluster_AA_model{model}.fasta\"),\"w\")\n",
    "SeqIO.write(records,handle,\"fasta\")\n",
    "\n",
    "handle.close()\n",
    "\n",
    "infile, alignfile = [\n",
    "        os.path.join(os.getcwd(), f\"Alphaproteobacteria_subcluster_AA_model{model}.fasta\"), \n",
    "        os.path.join(os.getcwd(), f\"Alphaproteobacteria_subcluster_AA_model{model}.afa\")]\n",
    "print( infile, \"\\n\", alignfile)\n",
    "    #alignent\n",
    "!mafft-linsi {infile}>{alignfile}\n",
    "\n",
    "records = []\n",
    "for seq_record in SeqIO.parse(f\"Alphaproteobacteria_subcluster_AA_model{model}.afa\", \"fasta\"):\n",
    "   \n",
    "    records.append(seq_record)\n",
    "\n",
    "handle = open(os.path.join(f\"Alphaproteobacteria_subcluster_AA_model{model}.cl\"),\"w\")\n",
    "SeqIO.write(records,handle,\"clustal\")\n",
    "\n",
    "handle.close()\n",
    "    # model12.append(str(seq_record.id).replace(\"-1\", \".1\").replace(\"-2\", \".2\"))\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "b6980814-9a2b-44e9-8d6f-2101ba1b7bb8",
   "metadata": {},
   "outputs": [],
   "source": [
    "records = []\n",
    "for seq_record in SeqIO.parse(os.path.join(\"Alphaproteobacteria_subcluster_CDS_model4.fasta\"), \"fasta\"):\n",
    "    seq_record.seq = seq_record.seq[0:330]\n",
    "    if str(seq_record.id)  not in [\n",
    "    #     \"GCF_001687365-1-ADU59_RS12685\", # related to PP_1066\n",
    "    #     \"GCF_002208825-2-CEP68_RS17635\", # related to PP_1066\n",
    "    #     \"GCF_001870665-2-BG454_RS16400\", #related to PP_0263\n",
    "    #     \"GCF_001687365-1-ADU59_RS19520\",  #related to PP_0263\n",
    "    #     \"GCF_000421265-1-K355_RS0119855\", #related to PP_0263\n",
    "    #     \"GCF_000374005-1-B152_RS0108145\", #related to PP_0263\n",
    "    #     \"GCF_000423645-1-G568_RS0115095\", #related to PP_1401\n",
    "    #     \"GCF_900110775-1-BSF79_RS15930\", #related to PP_1401\n",
    "    #     \"GCF_000169415-1-SSE37_RS16260\", #related to PP_1401\n",
    "    #     \"GCF_000442315-1-RUTHE_RS05685\", #related to PP_1401\n",
    "    #     \"GCF_001870665-2-BG454_RS09765\", #related to PP_1401\n",
    "        \"GCF_000421265.1-K355_RS0117460\", #possible outgroup (2)\n",
    "        \"GCF_000423645-1-G568_RS0108480\", #possible outgroup (1)\n",
    "        \"GCF_000169415-1-SSE37_RS08100\", #possible outgroup (1)\n",
    "        \"GCF_900110775-1-BSF79_RS18365\", #possible outgroup (1)\n",
    "    ]: \n",
    "        records.append(seq_record)\n",
    "\n",
    "\n",
    "handle = open(\"Alphaproteobacteria_subcluster_CDS_REC_model10.fasta\",\"w\")\n",
    "SeqIO.write(records,handle,\"fasta\")\n",
    "handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "7fcc71df-a77c-4fd5-9d3e-f3af0bd9ccb8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">GCF_000021325-1-GDIA_RS09815\n",
      "ATGAGCGGGCAGGCACCGGAAGACCGGCCCCGGCCGCATGTCGTCGTCGTCGATGACGAT\n",
      "GCCGGGGTGCGCGCGTCGCTGGACAGCCTGTTCCGTTCCGCCGGCTTCACCGTGACCCTG\n",
      "TTTGCCGACCCGCACGCCTTCCTGAAGGCCGGCGTGCCACAGGGGCCGTGCTGCCTGGTA\n",
      "CTGGACGTGAAGCTGGGCCGGATCGACGGCCTCGACTTCCACGAGGCGATCGTCGACGCG\n",
      "GGACTCCGCATCCCGACCATCATCATGACCGGTCATGGGGACATTCCGATGGCCGTCCGG\n",
      "GGCATGCGGACGGGGGCGATCGATTTCCTG\n",
      ">GCF_000017565-1-PLAV_RS12075\n",
      "ATGCTCGCGAGTGCAATGGTGGCGGTTGTCGATGACGATGAATCGGCACGTGAGGCGATA\n",
      "GTCGAGCTGGTGAGGGCTTTGGGATACGAAGCTCTCGGCTTTTCAAGCGGAGAGAATTTT\n"
     ]
    }
   ],
   "source": [
    "!head Alphaproteobacteria_subcluster_CDS_REC_model10.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "7e156e86-3666-4ad1-bab3-fd537c4a38b8",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.fasta \n",
      " /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa\n",
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "All-to-all alignment.\n",
      "tbfast-pair (nuc) Version 7.419\n",
      "alg=L, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "   20 / 25\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP     9 /24 \n",
      "Reallocating..done. *alloclen = 1669\n",
      "STEP    24 /24 \n",
      "done.\n",
      "tbfast (nuc) Version 7.419\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "   20 / 25\n",
      "Segment   1/  1    1- 480\n",
      "STEP 003-014-1  rejected..    identical.    identical.    identical.    accepted. identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    rejected. rejected. rejected.\n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.419\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "IQ-TREE multicore version 1.6.12 for Linux 64-bit built Aug 15 2019\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\n",
      "Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\n",
      "\n",
      "Host:    jupyter (AVX, FMA3, 503 GB RAM)\n",
      "Command: /opt/iqtree/iqtree-1.6.12-Linux/bin/iqtree -s /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa -m TVM+F+I+G4 -redo -bb 1000\n",
      "Seed:    698443 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Fri Aug 19 10:13:04 2022\n",
      "Kernel:  AVX+FMA - 1 threads (64 CPU cores detected)\n",
      "\n",
      "HINT: Use -nt option to specify number of threads because your CPU has 64 cores!\n",
      "HINT: -nt AUTO will automatically determine the best number of threads to use.\n",
      "\n",
      "Reading alignment file /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 25 sequences with 462 columns, 348 distinct patterns\n",
      "277 parsimony-informative, 55 singleton sites, 130 constant sites\n",
      "                                Gap/Ambiguity  Composition  p-value\n",
      "   1  GCF_000021325-1-GDIA_RS09815     28.57%    failed      0.96%\n",
      "   2  GCF_000017565-1-PLAV_RS12075     28.57%    passed     98.37%\n",
      "   3  GCF_000017565-1-PLAV_RS12070     28.57%    passed     25.07%\n",
      "   4  GCF_000017565-1-PLAV_RS11965     28.57%    passed     56.59%\n",
      "   5  GCF_000017565-1-PLAV_RS11955     28.57%    passed     53.05%\n",
      "   6  GCF_000017565-1-PLAV_RS10335     28.57%    passed     90.08%\n",
      "   7  GCF_000017565-1-PLAV_RS05525     28.57%    passed     23.35%\n",
      "   8  GCF_001687365-1-ADU59_RS05750    28.57%    passed     65.42%\n",
      "   9  GCF_000385335-1-A3OQ_RS0103935   28.57%    passed     99.23%\n",
      "  10  GCF_900168195-1-B5Y19_RS10715    28.57%    failed      3.03%\n",
      "  11  GCF_900168195-1-B5Y19_RS10675    28.57%    passed     35.59%\n",
      "  12  GCF_900168195-1-B5Y19_RS10665    28.57%    passed     37.46%\n",
      "  13  GCF_000374005-1-B152_RS0109090   28.57%    failed      0.11%\n",
      "  14  GCF_002208825-2-CEP68_RS27755    28.57%    passed     30.28%\n",
      "  15  GCF_000423645-1-G568_RS0115095   28.57%    passed     51.15%\n",
      "  16  GCF_900110775-1-BSF79_RS15930    28.57%    passed     11.15%\n",
      "  17  GCF_000169415-1-SSE37_RS16260    28.57%    passed     76.96%\n",
      "  18  GCF_000442315-1-RUTHE_RS05685    28.57%    passed     79.14%\n",
      "  19  GCF_001870665-2-BG454_RS09765    28.57%    passed     59.38%\n",
      "  20  GCF_002208825-2-CEP68_RS17635    28.57%    passed      8.51%\n",
      "  21  GCF_001687365-1-ADU59_RS12685    28.57%    passed     16.13%\n",
      "  22  GCF_001870665-2-BG454_RS16400    28.57%    passed     62.63%\n",
      "  23  GCF_001687365-1-ADU59_RS19520    28.57%    passed     96.40%\n",
      "  24  GCF_000421265-1-K355_RS0119855   28.57%    passed     20.05%\n",
      "  25  GCF_000374005-1-B152_RS0108145   28.57%    failed      0.00%\n",
      "****  TOTAL                            28.57%  4 sequences failed composition chi2 test (p-value<5%; df=3)\n",
      "\n",
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.002 seconds\n",
      "Generating 1000 samples for ultrafast bootstrap (seed: 698443)...\n",
      "\n",
      "NOTE: 2 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "Thoroughly optimizing +I+G parameters from 10 start values...\n",
      "Init pinv, alpha: 0.000, 1.000 / Estimate: 0.000, 0.907 / LogL: -7782.899\n",
      "Init pinv, alpha: 0.031, 1.000 / Estimate: 0.093, 1.690 / LogL: -7768.936\n",
      "Init pinv, alpha: 0.063, 1.000 / Estimate: 0.092, 1.695 / LogL: -7768.934\n",
      "Init pinv, alpha: 0.094, 1.000 / Estimate: 0.093, 1.709 / LogL: -7768.926\n",
      "Init pinv, alpha: 0.125, 1.000 / Estimate: 0.095, 1.726 / LogL: -7768.929\n",
      "Init pinv, alpha: 0.156, 1.000 / Estimate: 0.096, 1.736 / LogL: -7768.937\n",
      "Init pinv, alpha: 0.188, 1.000 / Estimate: 0.095, 1.727 / LogL: -7768.928\n",
      "Init pinv, alpha: 0.219, 1.000 / Estimate: 0.095, 1.728 / LogL: -7768.930\n",
      "Init pinv, alpha: 0.250, 1.000 / Estimate: 0.096, 1.729 / LogL: -7768.932\n",
      "Init pinv, alpha: 0.281, 1.000 / Estimate: 0.096, 1.729 / LogL: -7768.935\n",
      "Optimal pinv,alpha: 0.093, 1.709 / LogL: -7768.926\n",
      "\n",
      "Parameters optimization took 1.365 sec\n",
      "Computing ML distances based on estimated model parameters... 0.008 sec\n",
      "Computing BIONJ tree...\n",
      "0.001 seconds\n",
      "Log-likelihood of BIONJ tree: -7778.052\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 0.236 second\n",
      "Computing log-likelihood of 98 initial trees ... 0.919 seconds\n",
      "Current best score: -7767.138\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -7761.309\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 3: -7759.209\n",
      "Iteration 10 / LogL: -7759.439 / Time: 0h:0m:3s\n",
      "Iteration 20 / LogL: -7759.430 / Time: 0h:0m:4s\n",
      "Finish initializing candidate tree set (11)\n",
      "Current best tree score: -7759.209 / CPU time: 3.205\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Iteration 30 / LogL: -7761.349 / Time: 0h:0m:5s (0h:0m:14s left)\n",
      "Iteration 40 / LogL: -7759.461 / Time: 0h:0m:6s (0h:0m:11s left)\n",
      "Iteration 50 / LogL: -7759.255 / Time: 0h:0m:8s (0h:0m:8s left)\n",
      "Log-likelihood cutoff on original alignment: -7793.207\n",
      "Iteration 60 / LogL: -7760.242 / Time: 0h:0m:9s (0h:0m:6s left)\n",
      "Iteration 70 / LogL: -7760.261 / Time: 0h:0m:10s (0h:0m:4s left)\n",
      "Iteration 80 / LogL: -7760.235 / Time: 0h:0m:11s (0h:0m:3s left)\n",
      "Iteration 90 / LogL: -7764.181 / Time: 0h:0m:12s (0h:0m:1s left)\n",
      "Iteration 100 / LogL: -7761.358 / Time: 0h:0m:13s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -7793.207\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.999\n",
      "TREE SEARCH COMPLETED AFTER 104 ITERATIONS / Time: 0h:0m:14s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -7759.209\n",
      "Optimal log-likelihood: -7759.209\n",
      "Rate parameters:  A-C: 3.50179  A-G: 5.21125  A-T: 2.17031  C-G: 4.76460  C-T: 5.21125  G-T: 1.00000\n",
      "Base frequencies:  A: 0.198  C: 0.291  G: 0.305  T: 0.207\n",
      "Proportion of invariable sites: 0.093\n",
      "Gamma shape alpha: 1.682\n",
      "Parameters optimization took 1 rounds (0.022 sec)\n",
      "BEST SCORE FOUND : -7759.209\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.splits.nex\n",
      "Total tree length: 11.614\n",
      "\n",
      "Total number of iterations: 104\n",
      "CPU time used for tree search: 12.699 sec (0h:0m:12s)\n",
      "Wall-clock time used for tree search: 12.716 sec (0h:0m:12s)\n",
      "Total CPU time used: 14.212 sec (0h:0m:14s)\n",
      "Total wall-clock time used: 14.232 sec (0h:0m:14s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.splits.nex...\n",
      "25 taxa and 226 splits.\n",
      "Consensus tree written to /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.contree\n",
      "Reading input trees file /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.contree\n",
      "Log-likelihood of consensus tree: -7759.406\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.iqtree\n",
      "  Maximum-likelihood tree:       /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.treefile\n",
      "  Likelihood distances:          /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.splits.nex\n",
      "  Consensus tree:                /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.contree\n",
      "  Screen log file:               /opt/jupyterhub/shared_notebooks/m-group/ALI/MG_stuff/RR_bioinformatics/TCSplayground/Alphaproteobacteria_subcluster/CDS/Alphaproteobacteria_subcluster_CDS_REC_model10.afa.log\n",
      "\n",
      "Date and Time: Fri Aug 19 10:13:19 2022\n"
     ]
    }
   ],
   "source": [
    "for model in [\n",
    "    10,\n",
    "    # 9, \n",
    "    # 7,\n",
    "    # 4,\n",
    "    # 1,2,3,5,\n",
    "    # 14,\n",
    "    # 6,8,12,13\n",
    "             ]:\n",
    "     #file assignment\n",
    "    infile, alignfile = [\n",
    "        os.path.join(os.getcwd(), f\"Alphaproteobacteria_subcluster_CDS_REC_model{model}.fasta\"), \n",
    "        os.path.join(os.getcwd(), f\"Alphaproteobacteria_subcluster_CDS_REC_model{model}.afa\")]\n",
    "    print( infile, \"\\n\", alignfile)\n",
    "        #alignent\n",
    "    !mafft-linsi {infile}>{alignfile}\n",
    "     #iqtree\n",
    "    !/opt/iqtree/iqtree-1.6.12-Linux/bin/iqtree -s {alignfile} -m TVM+F+I+G4 -o GCF_000169415-1-SSE37_RS08100 -redo -bb 1000\n",
    "    \n",
    "     #file cleanup\n",
    "    iqfiles = os.path.join(os.getcwd(), f\"Alphaproteobacteria_subcluster_CDS_REC_model{model}.afa.*\")\n",
    "    directory = os.path.join(os.getcwd(), \"iqtree_Results\", f\"Model_{model}\")\n",
    "    if not os.path.isdir(directory):\n",
    "        print(directory)\n",
    "\n",
    "        os.mkdir(directory)\n",
    "    !mv {iqfiles} {directory}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "b7682975-51a6-42f3-a13d-2dd018b4367e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#file cleanup\n",
    "for model in [\n",
    "    9, 7,\n",
    "    # 4\n",
    "    # 1,2,3,5,\n",
    "    # 14,\n",
    "    # 6,8,12,13\n",
    "             ]:\n",
    "    file = os.path.join(os.getcwd(), f\"Alphaproteobacteria_subcluster_CDS_REC_model{model}.afa.*\")\n",
    "    directory = os.path.join(os.getcwd(), \"iqtree_Results\", f\"Model_{model}\")\n",
    "    if not os.path.isdir(directory):\n",
    "        print(directory)\n",
    "\n",
    "        os.mkdir(directory)\n",
    "    !mv {file} {directory}\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c5f56122-86f5-4185-aaf2-79f4fe7f0440",
   "metadata": {},
   "source": [
    "#### Proceeding with model 12\n",
    "- model 12 has bootstrap value of 100 at partition of interest\n",
    "- Model 12 agrees with model 7 with regard to partition of AAA+ paralogs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9738a7a4-0798-4cee-8a52-f8e918c21a19",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Biopython 3.8",
   "language": "python",
   "name": "biopython_3.8"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
